hist/hist/test/test_TH1_SaveAs.cxx

```diff
@@ -0,0 +1,224 @@
+#include "gtest/gtest.h"
+
+#include "TString.h"
+#include "TH1.h"
+#include "TSystem.h"
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+struct TestSaveAs {
+   const Int_t Nbins = 5;
+   const Int_t N = Nbins+2;  // 0 and N are the under/overflow bins, resp.
+   TString fnam = "H1dump.";
+   Char_t filename[20];
```

Comment:
I would avoid old-style arrays of chars unless necessary, see comments in the single functions for improvement suggestions

Reply:
Ok!